src/include/txn.i

```diff
@@ -1091,12 +1107,24 @@ __wt_txn_update_check(WT_SESSION_IMPL *session, WT_UPDATE *upd)
         if (upd->txnid != WT_TXN_ABORTED) {
             if (ignore_prepare_set)
                 F_SET(txn, WT_TXN_IGNORE_PREPARE);
-            WT_STAT_CONN_INCR(session, txn_update_conflict);
-            WT_STAT_DATA_INCR(session, txn_update_conflict);
-            return (__wt_txn_rollback_required(session, "conflict between concurrent operations"));
+            return (__txn_rollback_required(session));
         }
     }
 
+    /* If there is no cursor, we don't need to check the on page value. */
+    WT_ASSERT(session, cbt != NULL || (cbt == NULL && !check_onpage_value));
+
+    /* Check conflict against the on page value if it exists. */
+    if (check_onpage_value) {
+        WT_RET(__wt_value_return_buf(cbt, cbt->ref, buf, &start, &stop));
```

Comment:
Reuse __wt_value_return_buf to crack the cell. But we don't need the data here. Pass a null to the function to tell it not returning any data.

I don't like this approach but it is easy so I did this. May be in the future we should refactor this and have a separate function to just unpack the cell and read the timestamps.